### PR TITLE
Add Do Not Sell privacy setting

### PR DESCRIPTION
**Pull request by robertsreberski**
#### Proposed Changes

- Add the Do Not Sell setting to the Privacy page (`/me/privacy/`)
  - Disable setting if cookie banner has not been yet accepted
  - Don't show the setting for non-CCPA regions 

#### Testing Instructions

* Checkout changes and run Calypso dev env
* Go to `calypso.localhost:3000/me/privacy?flags=cookie-banner`
* Set cookies for CCPA region: `country_code=US; region=California;`
* Set all buckets to accepted: `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Atrue%7D%7D';`
* You should be able to see the Do Not Sell setting on Privacy page
CleanShot 2022-11-29 at 18 27 22@2x

* You should be able to switch Do Not Sell setting - it should impact `buckets.advertising` value in `sensitive_pixel_options` cookie
* Remove the `sensitive_pixel_options` cookie (stay in CCPA region)
* You should see Do Not Sell setting on Privacy page, but the switch should be disabled
CleanShot 2022-11-29 at 18 26 23@2x

* Change `country_code` to any other country e.g. `country_code=NL`
* The Do Not Sell setting should not appear on Privacy page

### CODE REVIEW

File: client/me/privacy/do-not-sell.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+					) }
+				</p>
+				<p>
+					{ translate(
+						'Like many websites, we share some of the data we collect through cookies with certain third party advertising and analytics vendors. The personal information we share includes online identifiers; internet or other network or device activity (such as cookie information, other device identifiers, and IP address); and geolocation data (approximate location information from your IP address). We do not share information that identifies you personally, like your name or contact information.'
+					) }
+				</p>
+				<p>
+					{ translate(
+						'We never directly sell your personal information in the conventional sense (i.e., for money), but in some U.S. states the sharing of your information with advertising/analytics vendors can be considered a “sale” of your information, which you may have the right to opt out of. To opt out, click the link below.'
```
Comment (IanRamosC) [inline client/me/privacy/do-not-sell.jsx:86]: This isn't directly related to the code, but to the choice of words in "To opt out, click the link below". The switch is not really a link, and the only link below that specific phrase is the one to `contact@automattic.com`. Besides that, everything else looks good.

**Review by IanRamosC — COMMENTED** [review]
Code looks good and it tests well. Tried all the different cookie settings and it worked as expected. Left one comment about one minor thing, but other than that I believe we are good to go.

File: client/me/privacy/do-not-sell.jsx (hunk 1 above)
Comment (robertsreberski) [reply client/me/privacy/do-not-sell.jsx:86]: @gmovr I've taken this text from a8c-analytics.js PR - is it up to date with legal?

Comment (gmovr) [reply client/me/privacy/do-not-sell.jsx:86]: Yes. We can get a new revision, but I wouldn't consider it a blocker for this. To get it out before EOY, and not having to wait for new translations.

**Comment by gmovr** [comment]
Code looks good, and it works following the testing instructions. Ref our meeting earlier, I don't think we need to wait for the cookie before allowing them to opt-out. Since advertising trackers will be opt-out in certain US states, while they are opt-in in GDPR. :) Let's clarify this, and then I'm happy to approve this.

**Review by gmovr — APPROVED** [review]
I rebased this on top of https://github.com/Automattic/wp-calypso/pull/70833 -- and the preferences page shows correctly. 

To merge this, we still depend on updating the tracking buckets for the opt-out to fully work. E.g. if I set my region to `California` no trackers are fired, while if I set it to `New York` advertising trackers are firing as default behavior. With the opt-out in "CCPA regions" advertising trackers should fire as default, until the user opts out. :) 

I'll approve this, the cookie is set correctly. I'll add the "Do not merge" label, until the other PRs are across the finish line (and this PR depends on those).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7664393

Thank you @robertsreberski for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7664393).

File: client/me/privacy/do-not-sell.jsx (hunk 1 above)
Comment (kbrown9) [reply client/me/privacy/do-not-sell.jsx:86]: Have there been any updates on the new text from legal? I think most site visitors can figure out that they just need to use the toggle switch, but the phrase about clicking the link below is confusing.

**Pull request merged**